core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,368 @@
+  };
+
+  this.publishNewBlogPost = async function(
+      blogPostTitle, richTextInstructions, tags) {
+    await action.sendKeys(
+      'New blog post title field', blogPostTitleFieldElement, blogPostTitle);
+    await this.selectTags(tags);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await this.setThumbnailImage();
+    await action.click('Publish Blog Post Button', publishBlogPostButton);
+    await action.click(
+      'Confirm Publish Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast(
+      'Blog Post Saved and Published Succesfully.');
+  };
+
+  this.publishDraftBlogPost = async function(tags) {
+    await this.selectTags(tags);
+    await this.setThumbnailImage();
+    await action.click('Publish Blog Post Button', publishBlogPostButton);
+    await action.click(
+      'Confirm Publish Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast(
+      'Blog Post Saved and Published Succesfully.');
+  };
+
+  this.selectTags = async function(tags) {
+    for (i = 0; i < await blogPostTags.count(); i++) {
+      var tag = blogPostTags.get(i);
+      var tagName = await action.getText(
+        `Blog Post Editor tag ${i}`, tag);
+      if (tags.includes(tagName)) {
+        await action.click(
+          'select blog post tag', tag);
+      }
+    }
+  };
+
+  this.getMatTab = async function(tabName) {
+    for (i = 0; i < await matTabLabels.count(); i++) {
+      var matTab = matTabLabels.get(i);
+      var tabText = await action.getText(
+        `Blog Dashboard tab ${i}`, matTab);
+      if (tabText.startsWith(tabName)) {
+        await action.click(`${tabName} tab`, matTab);
+      }
+    }
+  };
+
+  this.navigateToPublishTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await this.getMatTab('PUBLISHED');
+  };
+
+  this.navigateToDraftsTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await this.getMatTab('DRAFTS');
+  };
+
+  this.expectNumberOfDraftBlogPostsToBe = async function(number) {
+    await this.waitForDraftsBlogPostsToLoad();
+    expect(await draftBlogPostTiles.count()).toBe(number);
+  };
+
+  this.blogDashboardIntroMessageIsVisible = async function() {
+    await waitFor.visibilityOf(
+      blogDashboardIntroMessageContainer, 'Blog Dashboard Intro message ' +
+      'taking too long to be visible');
+  };
+
+  this.expectNumberOfPublishedBlogPostsToBe = async function(number) {
+    await this.waitForPublishedBlogPostsToLoad();
+    expect(await publishedBlogPostTiles.count()).toBe(number);
+  };
+
+  this.expectNumberOfBlogPostsRowsToBe = async function(number) {
+    await this.waitForDraftsBlogPostsToLoad();
+    expect(await blogPostListItems.count()).toBe(number);
+  };
+
+  this.getBlogPostTileEditOption = async function(title) {
+    for (i = 0; i < await blogPostTiles.count(); i++) {
```

Comment:
Can you wait for the items to load? Be careful to handle the case where there are no items if applicable

Reply:
Done.

Reply:
Done.